src/WorkerPool.js

```diff
@@ -1,25 +1,46 @@
+const onErr = err => {
+	console.log('err', err);
+};
 
 module.exports = class WorkerPool {
-	constructor () {
-		this.workers = {};
+	constructor (url, max = 4) {
+		this.workers = [];
+		this.url = url;
+		this.max = max;
 	}
 
-	getWorker (url) {
-		if (!this.workers[url]) {
-			this.workers[url] = [];
-		}
-
-		if (this.workers[url].length === 0) {
-			let worker = new Worker(url);
-			this.workers[url].push(worker);
+	getWorker () {
+		if (this.workers.length === 0) {
+			let worker = new Worker(this.url);
+			worker.addEventListener('error', onErr);
+			this.workers.push(worker);
 		}
 
-		let worker = this.workers[url].pop();
+		let worker = this.workers.pop();
 
 		return worker;
 	}
 
-	returnWorker (url, worker) {
-		this.workers[url].push(worker);
+	returnWorker (worker) {
+		if (this.workers.length >= this.max) {
```

Comment:
I think this would be better implemented in a different way? It might be hard to debug it if the worker is accidentally added to the wrong worker list.

```javascript
worker.returnToWorkers = function () {
   // Return it to the list
}
```
might be better?

Reply:
the key here is just that since there is only a single list of workers, so it would never get added to the wrong one

Reply:
If there is only one "list" then the url shouldn't be part of the constructor, no?

Reply:
so there is only one list per worker pool, but at some latter date we might want to have multiple worker pools so having it as a parameter makes it more flexible in the future, also we could hard code it but it doesn't have access to context so we'd probably have to pass that in anyway.